Tests/Shared.Specs/ExceptionAssertionSpecs.cs

```diff
@@ -889,12 +888,132 @@ public void When_no_exception_should_be_thrown_and_none_was_it_should_not_throw(
+                .WithMessage("* value of pollInterval must be positive*");
+        }
+
+        [Fact]
+        public void NotThrowAfter_when_no_exception_should_be_thrown_after_wait_time_but_it_was_it_should_throw()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            var watch = Stopwatch.StartNew();
+            var waitTime = 100.Milliseconds();
+            var pollInterval = 10.Milliseconds();
+
+            Action ThrowLongerThanWaitTime = () =>
+                {
+                    if (watch.Elapsed <= waitTime + (waitTime.Milliseconds / 2).Milliseconds())
+                        throw new ArgumentException("An exception was forced");
+                };
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action action = () =>
+                ThrowLongerThanWaitTime.Should().NotThrowAfter(waitTime, pollInterval, "we passed valid arguments");
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            action.Should().Throw<XunitException>()
+                         .WithMessage("Did not expect any exception after 0.100s because we passed valid arguments*");
+        }
+
+        [Fact]
+        public void NotThrowAfter_when_no_exception_should_be_thrown_after_wait_time_and_none_was_it_should_not_throw()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            var watch = Stopwatch.StartNew();
+            var waitTime = 100.Milliseconds();
+            var pollInterval = 10.Milliseconds();
+
+            Action ThrowShorterThanWaitTime = () =>
+                {
+                    if (watch.Elapsed <= (waitTime.Milliseconds / 2).Milliseconds())
+                        throw new ArgumentException("An exception was forced");
+                };
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act / Assert
+            //-----------------------------------------------------------------------------------------------------------
+            ThrowShorterThanWaitTime.Should().NotThrowAfter(waitTime, pollInterval);
```

Comment:
🔧 The entire test suite might not be consistent about it, but to adhere to the AAA-style you could write

```
// Act
Action act = () => ThrowShorterThanWaitTime.Should().NotThrowAfter(waitTime, pollInterval);

// Assert
act.Should().NotThrow();
```